### PR TITLE
Refactor shared configs

### LINKED ISSUE #5: Extend repository to include more shared configs
### What would you like?

I would like to extend the repository to also include shared configs for other things than just `eslint`. For example `husky`, `releaserc` and `lintstaged`. If not for fully preset configs, at least a way to structurally extend a solid base.

### Why is this needed?

Brings more structure and simplicity to other repositories. Leads to a default being used for the whole organisation, and easy changes across the whole organisation.

### Other information

_No response_

### PR DESCRIPTION
**Pull request by Gijsdeman**
# Description
Splits the configurations in separate packages. Allows for more fine-grained control and generalized the repository to allow for more different shared configurations to be added.

## Related issues/external references
Closes #5.

## Types of changes
- Breaking change _(fix or feature that would cause existing functionality to change)_
- Documentation improvement

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,66 +1,37 @@
 # GEWIS Eslint Config
 
-This repository contains a shared ESLint configuration for use across all GEWIS projects. It aims to enforce consistent coding standards and code quality.
+This repository contains a shared configurations for use across all GEWIS projects. It aims to enforce consistent coding standards and code quality.
```
Comment (JustSamuel) [inline README.md:3]: > a shared

File: README.md [hunk 2]
```diff
@@ -1,66 +1,37 @@
 # GEWIS Eslint Config
```
Comment (JustSamuel) [inline README.md:1]: GEWIS JavaScript Configs

File: README.md [hunk 3]
```diff
@@ -1,66 +1,37 @@
...
-In your project's root directory, create or update the `eslint.config.{js,mjs,cjs,ts,mts,cts}` file to extend this configuration. 
+### ESLint 
+In your project's root directory, create or update the `eslint.config.{js,mjs,cjs,ts,mts,cts}` file to extend this configuration.
 You can pick a combination of various configuration files:
 
-- `eslint.common.mjs`
+- `index.mjs`
 - `eslint.vue.mjs`
-- `eslint.prettier.mjs`
+- `index.mjs`
```
Comment (JustSamuel) [inline README.md:17]: double

File: README.md [hunk 4]
```diff
@@ -1,66 +1,37 @@
...
+### Prettier
 
-/**
- * @type {import("prettier").Config}
- */
-export default {
-  ...prettier,
-};
-```
+In your project's root directory, create or update the `.prettierrc.mjs` file to extend this configuration.
```
Comment (JustSamuel) [inline README.md:28]: minimal example? or link to the readme

**Review by JustSamuel — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (Gijsdeman) [reply README.md:27]: No thank you, then you will have to keep those references up to date down the line.

**Review by JustSamuel — APPROVED** [review]

**Pull request merged**